docs/pages/experiments/material-ui/css-variables-custom-theme.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import {
+  CssVarsProvider,
+  useColorScheme,
+  createTheme,
+  ThemeInput,
+  alpha,
+  darken,
+} from '@mui/material/styles';
+import Moon from '@mui/icons-material/DarkMode';
+import Sun from '@mui/icons-material/LightMode';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import { teal, deepOrange, orange, cyan } from '@mui/material/colors';
+
+const ColorSchemePicker = ({ modeChanged }: { modeChanged: (mode: 'light' | 'dark') => void }) => {
+  const { mode, setMode } = useColorScheme();
+  const [mounted, setMounted] = React.useState(false);
+  React.useEffect(() => {
+    setMounted(true);
+  }, []);
+  if (!mounted) {
+    return null;
+  }
+
+  return (
+    <Button
+      variant="outlined"
+      onClick={() => {
+        if (mode === 'light') {
+          setMode('dark');
+          modeChanged('dark');
+        } else {
+          setMode('light');
+          modeChanged('light');
+        }
+      }}
+    >
+      {mode === 'light' ? <Moon /> : <Sun />}
+    </Button>
+  );
+};
+
+const defaultTheme = createTheme({
+  palette: {
+    primary: teal,
+    secondary: deepOrange,
+  },
+});
+
+const darkTheme = createTheme({
+  palette: {
+    primary: cyan,
+    secondary: orange,
+  },
+});
+
+const themeWithColorSchemes = {
+  ...defaultTheme,
+  colorSchemes: {
+    light: {
+      palette: {
+        primary: defaultTheme.palette.primary,
+        secondary: defaultTheme.palette.secondary,
+      },
+    },
+    dark: {
+      palette: {
+        primary: defaultTheme.palette.primary,
+        secondary: defaultTheme.palette.secondary,
+      },
+    },
+  },
+};
+
+const darkThemeWithColorSchemes = {
+  ...darkTheme,
+  colorSchemes: {
+    light: {
+      palette: {
+        primary: defaultTheme.palette.primary,
+        secondary: defaultTheme.palette.secondary,
+      },
+    },
+    dark: {
+      palette: {
+        primary: darkTheme.palette.primary,
+        secondary: darkTheme.palette.secondary,
+      },
+    },
+  },
+};
+
+export default function Page() {
+  const [theme, setTheme] = React.useState<ThemeInput>(themeWithColorSchemes);
+
+  const modeChanged = (newMode: 'light' | 'dark') => {
+    setTheme(newMode === 'dark' ? darkThemeWithColorSchemes : themeWithColorSchemes);
+  };
```

Comment:
Not sure why you added this but from what I look, it is unnecessary.

Using `CssVarsProvider` shift the mental model from "switching between theme via JS" to "preparing all the color schemes upfront"

So you would do this instead.

```js
<CssVarsProvider theme={{
  colorSchemes: {
    light: {
      palette: {
        primary: teal,
        secondary: deepOrange,
      },
    },
    dark: {
      palette: {
        primary: cyan,
        secondary: orange,
      },
    }
  }
}}>
```